core/network/corda-interop-app/interop-contracts/src/main/kotlin/com/weaver/corda/app/interop/states/AssetClaimStatusState.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.weaver.corda.app.interop.states
+
+import com.weaver.corda.app.interop.contracts.AssetTransferContract
+import net.corda.core.contracts.BelongsToContract
+import net.corda.core.contracts.LinearState
+import net.corda.core.contracts.ContractState
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+import java.time.Instant
+import net.corda.core.contracts.StaticPointer
+
+/**
+ * The AssetClaimStatusState stores the details about the asset claim status in the remote/importing network for asset transfer.
+ *
+ * The AssetClaimStatusState is created after performing the asset claim in the remote/importing newtork during asset transfer.
+ * This state is queried by the local/exporting network to check if the asset is claimed in the remote/importing network. If the asset is
+ * NOT claimed in the remote/importing network, then the asset will be reclaimed in the local/exporting network after the pledge timeout.
+ *
+ * @property pledgeId The unique identifier representing the pledge on an asset for transfer, in the exporting n/w.
+ * @property assetDetails String containing the marshalled asset (using JSON encoding) being transferred.
+ * @property localNetworkID The id of the network into which the asset is claimed (i.e., importing n/w).
+ * @property remoteNetworkID The id of the network in which the asset is pledged (i.e., exporting n/w).
+ * @property recipient The owner of the asset after transfer.
+ * @property recipientCert Certificate of the asset recipient (in base64 format) in the importing network.
+ * @property claimStatus Boolean value to convey if asset is claimed in the remote/importing network or not.
+ * @property expiryTimeSecs The future time in epoch seconds before which the asset claim in the remote/importing network is to be performed.
+ * @property expirationStatus Boolean variable to convey if the time in the remote/importing network is past expiryTimeSecs or not.
+ * @property linearId The unique identifier for this state object in the vault.
+ */
+@BelongsToContract(AssetTransferContract::class)
+data class AssetClaimStatusState(
+    // Only if the pledge happens on a Corda network, the pledgeId can be used as the linearId by the importing Corda network.
+    // However, the pledge might happen on a Fabric network with asset being imported into a Corda network.
+    // In this scenario, the pledgeId generated by a Fabric network cannot be used as the linearId in Corda network.
```

Comment:
I'm not sure we need to distinguish the source platforms this way. We'll have to hardcode an exception then, right? Whether the foreign network is Corda or otherwise?

Reply:
We are not distinguishing the source platforms Rama. We are just storing the `pledge-id` explicitly in the state hence it can be unaware of the source platform.